java/server/src/org/openqa/selenium/remote/server/rest/ResultConfig.java

```diff
@@ -46,9 +52,101 @@
   private final HandlerFactory handlerFactory;
   private final DriverSessions sessions;
   private final Logger log;
-
+  private final Set<String> nonInvasiveCommands = new HashSet<String>() {{
+    add(GET_ALL_SESSIONS);
+    add(GET_CAPABILITIES);
+
+    add(STATUS);
+
+    add(ADD_COOKIE);
+    add(GET_ALL_COOKIES);
+    add(DELETE_COOKIE);
+    add(DELETE_ALL_COOKIES);
+
+    add(FIND_ELEMENT);
+    add(FIND_ELEMENTS);
+    add(FIND_CHILD_ELEMENT);
+    add(FIND_CHILD_ELEMENTS);
+
+    add(GET_CURRENT_WINDOW_HANDLE);
+    add(GET_WINDOW_HANDLES);
+
+    add(GET_CURRENT_CONTEXT_HANDLE);
+    add(GET_CONTEXT_HANDLES);
+
+    add(GET_ACTIVE_ELEMENT);
+
+    add(GET_CURRENT_URL);
+    add(GET_PAGE_SOURCE);
+    add(GET_TITLE);
+
+    add(GET_ELEMENT_TEXT);
+    add(GET_ELEMENT_TAG_NAME);
+    add(IS_ELEMENT_SELECTED);
+    add(IS_ELEMENT_ENABLED);
+    add(IS_ELEMENT_DISPLAYED);
+    add(GET_ELEMENT_LOCATION);
+    add(GET_ELEMENT_LOCATION_ONCE_SCROLLED_INTO_VIEW);
+    add(GET_ELEMENT_SIZE);
+    add(GET_ELEMENT_ATTRIBUTE);
+    add(GET_ELEMENT_VALUE_OF_CSS_PROPERTY);
+    add(ELEMENT_EQUALS);
+
+    add(SCREENSHOT);
+    add(ELEMENT_SCREENSHOT);
+
+    add(GET_ALERT_TEXT);
+
+    add(SET_TIMEOUT);
+    add(IMPLICITLY_WAIT);
+    add(SET_SCRIPT_TIMEOUT);
+
+    add(GET_LOCATION);
+    add(SET_LOCATION);
+    add(GET_APP_CACHE);
+    add(GET_APP_CACHE_STATUS);
+    add(CLEAR_APP_CACHE);
+    add(IS_BROWSER_ONLINE);
+    add(SET_BROWSER_ONLINE);
+
+    add(GET_LOCAL_STORAGE_ITEM);
+    add(GET_LOCAL_STORAGE_KEYS);
+    add(SET_LOCAL_STORAGE_ITEM);
+    add(REMOVE_LOCAL_STORAGE_ITEM);
+    add(CLEAR_LOCAL_STORAGE);
```

Comment:
I would not add any command that can change the local storage. You have no warranty that the JS is not polling/listening to local storage changes to do some app specific logic (like logging out the user) triggering a load page.